pkg/api/v1alpha1/cluster_types.go

```diff
@@ -228,14 +229,7 @@ func WorkerNodeGroupConfigurationsSliceEqual(a, b []WorkerNodeGroupConfiguration
 		return false
 	}
```

Comment:
I think we can simplify this logic
we can build a map[key]workernodegroup with `a`
Then iterate over b. We check every workernodegroup in `b` to see if it's in `a`. If it's not, return false. If it is, compare the two worker node groups (probably with a new method?)

Reply:
do you mean map[name]workernodegroup?

Reply:
so I get what you're saying but this already does what we need it to

Reply:
with the only modification being the addition of the `name` to the key generation method